src/controls/controls.actions.ts

```diff
@@ -268,138 +270,116 @@ import { renderCircleControl, renderSquareControl } from './controls.render';
   }
 
   /**
-   * Detect if the fabric object is flipped on one side.
-   * @param {fabric.Object} target
-   * @return {Boolean} true if one flip, but not two.
+   * @see https://github.com/fabricjs/fabric.js/pull/8380
    */
-  function targetHasOneFlip(target) {
-    return target.flipX !== target.flipY;
-  }
-
-  /**
-   * Utility function to compensate the scale factor when skew is applied on both axes
-   * @private
-   */
-  function compensateScaleForSkew(
-    target,
-    oppositeSkew,
-    scaleToCompensate,
-    axis,
-    reference
+  function skewObject(
+    axis: TAxis,
+    { target, ex, ey, originX, originY, skewX, skewY, flipX, flipY },
+    pointer: Point
   ) {
-    if (target[oppositeSkew] !== 0) {
-      var newDim = target._getTransformedDimensions()[axis];
-      var newValue = (reference / newDim) * target[scaleToCompensate];
-      target.set(scaleToCompensate, newValue);
-    }
+    const { counterAxis, skew: skewKey } = AXIS_KEYS[axis],
+      offset = pointer
+        .subtract(new Point(ex, ey))
+        .divide(new Point(target.scaleX, target.scaleY))[axis],
+      flip = new Point(flipX ? -1 : 1, flipY ? -1 : 1)[axis],
+      skewingStart = new Point(skewX, skewY)[axis],
+      skewingBefore = new Point(target.skewX, target.skewY)[axis],
+      shearingStart = Math.tan(degreesToRadians(skewingStart)),
+      counterOriginFactor = new Point(
+        target.resolveOriginX(originX),
+        target.resolveOriginY(originY)
```

Comment:
my PR is changing resolveOriginX and resolveOriginY away from methods ( since they don't use this ) and there is just an exported function ( that could be an util ) that is called resolveOrigin and takes indifferently originX or originY

Reply:
Decide what to merge first and I will fix in the other